include/random_walks/new_gaussian_accelerated_billiard_walk.hpp

```diff
@@ -0,0 +1,277 @@
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef RANDOM_WALKS_NEW_GAUSSIAN_ACCELERATED_BILLIARD_WALK_HPP
+#define RANDOM_WALKS_NEW_GAUSSIAN_ACCELERATED_BILLIARD_WALK_HPP
+
+#include <Eigen/Eigen>
+#include <cmath>
+#include <iomanip>
+#include "convex_bodies/orderpolytope.h"
+#include "convex_bodies/ellipsoid.h"
+#include "convex_bodies/ballintersectconvex.h"
+#include "generators/boost_random_number_generator.hpp"
+#include "sampling/ellipsoid.hpp"
+#include "random_walks/uniform_billiard_walk.hpp"
+
+#include "random_walks/compute_diameter.hpp"
+
+
+struct GABW
+{
+    GABW(double L)
+            :   param(L, true)
+    {}
+
+    GABW()
+            :   param(0, false)
+    {}
+
+    struct parameters
+    {
+        parameters(double L, bool set)
+                :   m_L(L), set_L(set)
+        {}
+        double m_L;
+        bool set_L;
+    };
+
+    struct update_parameters
+    {
+        update_parameters()
+                :   facet_prev(0), hit_ball(false), inner_vi_ak(0.0), ball_inner_norm(0.0)
+        {}
+        int facet_prev;
+        bool hit_ball;
+        double inner_vi_ak;
+        double ball_inner_norm;
+    };
+
+    parameters param;
+
+
+    template
+            <
+                    typename Polytope,
+                    typename RandomNumberGenerator
+            >
+    struct Walk
+    {
+        typedef typename Polytope::PointType Point;
+        typedef typename Polytope::MT MT;
+        typedef typename Polytope::VT VT;
+        typedef typename Point::FT NT;
+
+        template <typename GenericPolytope>
+        Walk(GenericPolytope& P,
+             Point const& p,
+             MT const& E,   // covariance matrix representing the Gaussian distribution
+             RandomNumberGenerator &rng)
+        {
+            _update_parameters = update_parameters();
+            _Len = compute_diameter<GenericPolytope>
+                ::template compute<NT>(P);
+
+            _AA.noalias() = P.get_mat() * P.get_mat().transpose();
+            _rho = 1000 * P.dimension(); // upper bound for the number of reflections (experimental)
+            initialize(P, p, E, rng);
+        }
+
+        template <typename GenericPolytope>
+        Walk(GenericPolytope& P,
+             Point const& p,
+             MT const& E,   // covariance matrix representing the Gaussian distribution
+             RandomNumberGenerator &rng,
+             parameters const& params)
+        {
+            _update_parameters = update_parameters();
+            _Len = params.set_L ? params.m_L
+                              : compute_diameter<GenericPolytope>
+                                ::template compute<NT>(P);
+            
+            _AA.noalias() = P.get_mat() * P.get_mat().transpose();
+            _rho = 1000 * P.dimension(); // upper bound for the number of reflections (experimental)
+            initialize(P, p, E, rng);
+        }
+
+        template <typename GenericPolytope>
```

Comment:
Mainly, the only different parts are the way the vector is chosen each time, and the reflection function, there are also a few added lines in the initialization for computing the AE and AEA matrices, and a few lines after each line_positive_intersect which uses _AA to adjust for the scaling coefficient. I don't really know how we could reuse parts of the code from uniform billiard walk, other than if we were to replace it. Since, in theory, this should be the same as uniform abw if the given covariance matrix is the identity, but that would change how the walk is called, and might introduce other complications